### ISSUE TITLE
Problem with lexographical sorting

### ISSUE THREAD

**Issue by leeprevost** [leeprevost/OrdinalClassifier#6]
I think I have a problem when I hand this classifier a y with text based classes (ie. low, medium, high).    The default sorter sorts this lexographically (low, high, medium) which makes medium act like a pos class.

sklearn has some PRs for this:
#https://github.com/scikit-learn/scikit-learn/issues/4450
#https://github.com/scikit-learn/scikit-learn/issues/13631

Looks like I need to:
- encode any ordinal informaiton from y
- don't default to "sorted" 
- label binarize y and keep separate classes_ array

I already allow for custom ordering. 

From testing on diabetes dataset, ordering does have a big effect on precision and recall of pos_class (ie. "high" disease progression in diabetes) so important to get this right.

LP
